lib/execution_segment.go

```diff
@@ -384,13 +385,82 @@ func (ess ExecutionSegmentSequence) String() string {
 	return strings.Join(result, ",")
 }
 
+func (ess ExecutionSegmentSequence) Len() int {
+	return len(ess)
+}
+
+func (ess ExecutionSegmentSequence) Less(i, j int) bool {
+	// Yes this Less is actually More, but we want it sorted in descending order and the alternative is to reverse it after sort
+	return ess[i].length.Cmp(ess[j].length) > 0
+}
+
+func (ess ExecutionSegmentSequence) Swap(i, j int) {
+	ess[i], ess[j] = ess[j], ess[i]
+}
+
+var _ sort.Interface = ExecutionSegmentSequence{}
+
+// lowest common denominator based on https://rosettacode.org/wiki/Least_common_multiple#Go
+func (ess ExecutionSegmentSequence) lcd() int64 {
+	var m, n, z big.Int
+	z = *ess[0].length.Denom()
+	for _, seg := range ess[1:] {
+		m = z
+		n = *seg.length.Denom()
```

Comment:
Do you think it's worth it to add an optimization here that if `m` is equal to `n` we should just continue? After all, for the majority of cases we're going to partition segments equally...

Reply:
done in d911aba